src/Dispatcher/DecoratedDispatcher.php

```diff
@@ -55,8 +55,11 @@ public function dispatch(ServerRequestInterface $request, ResponseInterface $res
 		if (isset($e)) {
 			// Trigger exception decorator
 			$response = $this->decoratorManager->decorateResponse(IDecorator::ON_DISPATCHER_EXCEPTION, $request, $response, ['exception' => $e]);
-			// If there's no decorator to handle this exception, throw again
-			if ($response === null) throw $e;
+			// Rethrow exception so WrappedDispatcher could log it and return response if possible
+			if ($response !== null) {
+				throw new SnapshotException($e, $request, $response);
```

Comment:
Tohle se mi nejak nezda.

1. Kdyz to bude SnapshotException, tak se throw $e zachova uplne stejne.
2. Kdyz nebude, tak se throw $e zachova uplne stejne.

Reply:
Bylo to zbytečné throw navíc nebo zbytečná přiřazení do proměnné, tohle mi dávalo větší smysl. Chceš to změnit?

Reply:
Ten kod se tu duplikuje.

Radek 46: vyhodi se SnapshotException, vytahnou se request, response a .
Radek 55: je $e vyplnena, ano je, je response vyplnena, ano je, vyhodi se znovu SnapshotException s tim stejnym, jenze tohle cele zajisti radek 62

Reply:
Nejde o tentýž snapshot. První se vrací z parent::dispatch a z něj je potřeba získat response a exception pro předání do DecoratorManager.
Nová SnapshotException se vyhazuje v případě, že DecoratorManager nevrátí prázdnou response, což se může stát. A nový snapshot se vytváří, protože decoratory response upravují.

Reply:
Trochu jsem to přepsal, ať je kód srozumitelnější https://github.com/apitte/core/pull/77/commits/fe1f9601a29f5fea9aa4e01fdb9e4eafb65332ef